tests/python/pytorch/graphbolt/test_item_sampler.py

```diff
@@ -610,3 +614,110 @@ def test_ItemSetDict_node_pairs_negative_dsts(batch_size, shuffle, drop_last):
     assert torch.all(src_ids[:-1] <= src_ids[1:]) is not shuffle
     assert torch.all(dst_ids[:-1] <= dst_ids[1:]) is not shuffle
     assert torch.all(negs_ids[:-1] <= negs_ids[1:]) is not shuffle
+
+
+def distributed_item_sampler_subprocess(
+    proc_id,
+    nprocs,
+    item_set,
+    num_ids,
+    batch_size,
+    shuffle,
+    drop_last,
+    even_inputs,
+):
+    dist.init_process_group(
+        backend="gloo",  # Use GLOO backend for CPU multiprocessing
+        init_method=f"file:///{os.path.join(os.getcwd(), 'dis_tempfile')}"
+        if platform == "win32"
+        else "tcp://127.0.0.1:12345",
+        world_size=nprocs,
+        rank=proc_id,
+    )
+
+    # Create a DistributedItemSampler.
+    item_sampler = gb.DistributedItemSampler(
+        item_set,
+        batch_size=batch_size,
+        shuffle=shuffle,
+        drop_last=drop_last,
+        even_inputs=even_inputs,
+    )
+    data_loader = gb.SingleProcessDataLoader(item_sampler)
+
+    # Count the numbers of items and batches.
+    num_items = 0
+    sampled_count = torch.zeros(num_ids, dtype=torch.int32)
+    for i in data_loader:
+        # Count how many times each item is sampled.
+        sampled_count[i.seed_nodes] += 1
+        num_items += i.seed_nodes.size(0)
+    num_batches = len(list(item_sampler))
+
+    # Calculate expected numbers of items and batches.
+    expected_num_items = num_ids // nprocs + (num_ids % nprocs > proc_id)
+    if drop_last and expected_num_items % batch_size > 0:
+        expected_num_items -= expected_num_items % batch_size
+    expected_num_batches = expected_num_items // batch_size + (
+        (not drop_last) and (expected_num_items % batch_size > 0)
+    )
+    if even_inputs:
```

Comment:
Consider to test whether this option can actually even the input in different cases.